agent/dns_test.go

```diff
@@ -1636,6 +1636,88 @@ func TestDNS_ConnectServiceLookup(t *testing.T) {
 	}
 }
 
+func TestDNS_IngressServiceLookup(t *testing.T) {
+	t.Parallel()
+
+	a := NewTestAgent(t, "")
+	defer a.Shutdown()
+	testrpc.WaitForLeader(t, a.RPC, "dc1")
+
+	// Register ingress-gateway service
+	{
```

Comment:
I haven't seen anonymous blocks used before in other projects. I noticed a few existing tests have them as well. What is their purpose?

Would it make sense to use `t.Run()` here so that failures in this block include the context in the failure message, which is being provided by the comment?

Reply:
It's mostly for organizing the code to avoid reusing things between RPCs or operations that take a pointer for the request/reply. It lets you reuse the same var names (like `out`) for different successive operations too. I think it's just a minor stylistic convention though.

Adding a `t.Run()` doesn't seem that useful here though, since we don't really expect these RPC calls to fail - they're just setup work. The code below, inside the for loop could go in a t.Run though, since the different FQDNs are each a test case.

Reply:
I think wrapping the different questions in a t.Run is definitely a good idea, pushed a commit to do that